gc/gc-base/src/main/java/org/projectnessie/gc/base/IdentifiedResultsRepo.java

```diff
@@ -0,0 +1,202 @@
+import org.projectnessie.model.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import scala.Tuple2;
+
+/** DDL + DML functionality for the "IdentifiedResult" table. */
+public final class IdentifiedResultsRepo {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(IdentifiedResultsRepo.class);
+
+  private static final String COL_GC_RUN_START = "gcRunStart";
+  private static final String COL_GC_RUN_ID = "gcRunId";
+  private static final String COL_CONTENT_ID = "contentId";
+  private static final String COL_CONTENT_TYPE = "contentType";
+  private static final String COL_SNAPSHOT_ID = "snapshotId";
+  private static final String COL_REFERENCE_NAME = "referenceName";
+  private static final String COL_HASH_ON_REFERENCE = "hashOnReference";
+
+  private final Schema icebergSchema =
+      new Schema(
+          Types.StructType.of(
+                  // GC run start timestamp.
+                  required(1, COL_GC_RUN_START, Types.TimestampType.withZone()),
+                  // GC run-ID.
+                  required(2, COL_GC_RUN_ID, Types.StringType.get()),
+                  // Nessie Content.id
+                  optional(3, COL_CONTENT_ID, Types.StringType.get()),
+                  // Nessie Content.type
+                  optional(4, COL_CONTENT_TYPE, Types.StringType.get()),
+                  // Iceberg Table/View Content's snapshot/version id.
+                  optional(5, COL_SNAPSHOT_ID, Types.LongType.get()),
+                  // Name of the reference via which the contentID was collected
+                  optional(6, COL_REFERENCE_NAME, Types.StringType.get()),
+                  // Hash of the reference via which the contentID was collected
+                  optional(7, COL_HASH_ON_REFERENCE, Types.StringType.get()))
+              .fields());
+
+  private final StructType schema = SparkSchemaUtil.convert(icebergSchema);
+
+  private final SparkSession sparkSession;
+  private final String catalogAndTableWithRefName;
+
+  public IdentifiedResultsRepo(
+      SparkSession sparkSession, String catalog, String gcRefName, String gcTableIdentifier) {
+    this.sparkSession = sparkSession;
+    this.catalogAndTableWithRefName = withRefName(catalog, gcTableIdentifier, gcRefName);
+    createTableIfAbsent(sparkSession, catalog, TableIdentifier.parse(gcTableIdentifier), gcRefName);
+  }
+
+  public StructType getSchema() {
+    return schema;
+  }
+
+  /**
+   * Collect the expired contents for the given run id as spark dataset.
+   *
+   * @param runId run id of completed identify task.
+   * @return spark dataset of row where each row is having the expired contents per content id per
+   *     reference.
+   */
+  public Dataset<Row> collectExpiredContentsAsDataSet(String runId) {
+    // Example Query:
+    // SELECT * FROM nessie.db1.`identified_results@someGcRef`
+    //    WHERE gcRunId = '9e809dc9-ac35-41c9-b1c0-5ebcf452ea2d'
+    return sql(
+        "SELECT * FROM %s WHERE %s = '%s'",
```

Comment:
do we need to do `SELECT *`? Shouldn't we pass column names explicitly and fetch only what's needed?

Reply:
yeah, but in case a new column is added to this table, we will fetch it and maybe not necessarily

Reply:
This table is purely for GC consumer, if new column is added then it is required for the consumer. 
I will reevaluate this again while adding GC consumer code.